README.rst

```diff
@@ -74,10 +74,10 @@ To something like this:
         return Ok(user)
 
     user_result = get_user_by_email(email)
-    if isinstance(user_result, Ok):
+    if isinstance(user_result, Ok): # or `is_ok(user_result)`
         # type(user_result.value) == User
         do_something(user_result.value)
-    else:
+    else: # or `elif is_err(user_result)`
```

Comment:
I notice the `if/else` pattern, which I'd expect to work in TypeScript doesn't work here. Maybe it's due to pyright not being as advanced as TypeScript's type checker?

```python
x: Result[float, str] = indirection()  # you need an indirection (function that returns a Result), otherwise pyright ignores the type annotation and uses the Ok or Err as a literal

if is_ok(x):
    return x   # Ok[float]
else:
    return x   # Err[str] | Ok[float]     but you'd expect it to resolve to Err[str]
```

Reply:
TypeScript reference: https://www.typescriptlang.org/play?#code/C4TwDgpgBAogTnKBeKBvKBrAlgOwCYBcUA5BAsVAL4BQokUA8hsmproSQPYYU13QAlCAGcArgBtgLRswA+sBNWoBjTjmFSswgPrcWACjhEhYyQEoiiLTOQA+KHAB02fMiQpi3YktXrN+LDgIZWAsNQMLKBMJKSR7fXQXDlJyKjMfNQ0oAA9jERiWdkDg0LV9MwBuJSwAMyh9LV0MfWyzMzRqKC6ciqgAej6ZakooCHFhaFRO7uzegYU4YaA
```typescript
type Err = { kind: 'err' }
type Ok = { kind: 'ok' }
type Result =  Ok | Err

const is_ok = (r: Result): r is Ok => r.kind === 'ok'

const indirection = (): Result => ({ kind: 'err' })

const x: Result = indirection();

if (is_ok(x)) {
    x; // Ok
} else {
    x; // Err
}
```

Reply:
I re-read the discussion over on #69, this was already mentioned here
https://github.com/rustedpy/result/issues/69#issuecomment-1151233994

And there's an upstream discussion in the python/typing repo https://github.com/python/typing/discussions/1013

So we can ignore this for now

Reply:
yeah, if they write and implement the new PEP, this would just be a matter of doc change. I hope they will